superset/assets/src/visualizations/nvd3_vis.js

```diff
@@ -351,6 +352,7 @@ export default function nvd3Vis(slice, payload) {
         chart.showLegend(false);
       } else {
         chart.showLegend(fd.show_legend);
+        chart.legend.margin({ top: 5, right: 30, left: 80, bottom: 30 });
```

Comment:
Should `bottom: 30` be different when the bar values are not shown? What was the original value? We're trying to make the most out of the real estate, `30` pixels is significant when the chart is displayed small in a dashboard...